lib/rubocop/cop/style/hash_except.rb

```diff
@@ -104,8 +103,6 @@ def bad_method?(method_name, block)
             end
           else
             bad_method_with_poro?(block) do |key_arg, send_node|
-              return false if method_name == :reject && block.body.method?(:!)
```

Comment:
This was removed because it's part of the semantic check (`semantically_except_method?`). Originally added in https://github.com/rubocop/rubocop/pull/13015.